### ISSUE TITLE
[Bug]: Query fails after updating replicas with load release because collection hasn't been loaded or has been released

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#16641]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220424-b82f6a1a
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.0.dev33
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

queryNode:
      replicas: 5
  config:
    common:
      retentionDuration: 40
    queryNode:
      gracefulTime: 0
```
```


### Current Behavior

1. Create collection with 2 shards and insert 3000 entities with flush
2. Load collection with 1 replica and query
3. Load collection with 2 replicas (There is no exception ?)
4. Release collection and load with 2 replicas
5. Query entities and get exception `<MilvusException: (code=1, message=no shard leaders available for channel: by-dev-rootcoord-dml_35_432744296733736961v1, leaders: [11], err: fail to Query, QueryNode ID = 11, reason=collection hasn't been loaded or has been released, collection id = 432744296733736961)>`

**Test case:**
```Python
   @pytest.mark.tags(CaseLabel.L3)
    def test_load_replica_change(self):
        # create, insert
        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
        df = cf.gen_default_dataframe_data()
        insert_res, _ = collection_w.insert(df)
        assert collection_w.num_entities == ct.default_nb

        collection_w.load(replica_number=1)
        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
        collection_w.load(replica_number=2)
        collection_w.get_replicas()

        collection_w.release()
        collection_w.load(replica_number=2)
        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
```
**Case running result:**
```
name>: collection_5OpNL3Ce
<partitions>: [{"name": "_default", "collection_name": "collection_5OpNL3Ce", "description": ""}]
<description>: 
<schema>: {
  auto_id: False
  description: 
  fields: [{
    name: int64
    description: 
    type: 5
    is_primary: True
    a......  (api_request.py:27)
[2022-04-24 16:21:25,630 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float                                       float_vector
0         0     0.0  [0.007436856784492043, 0.09394764771862305, 0....
1         1     1.0  [0.01739919203371784, 0.12882032073533647, 0.1...
2         2     2.0  [0.06114443129401205, 0.11334566764309541, 0.0...
3         3    ......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-24 16:21:26,170 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 432744296864546818)  (api_request.py:27)
[2022-04-24 16:21:29,331 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:55)
[2022-04-24 16:21:30,485 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-24 16:21:30,485 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', None, None, 20], kwargs: {} (api_request.py:55)
[2022-04-24 16:21:30,810 - DEBUG - ci_test]: (api_response) : [{'int64': 0}]  (api_request.py:27)
[2022-04-24 16:21:30,810 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 2, 20], kwargs: {} (api_request.py:55)
[2022-04-24 16:21:30,839 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-24 16:21:30,840 - DEBUG - ci_test]: (api_request)  : [Collection.get_replicas] args: [20], kwargs: {} (api_request.py:55)
[2022-04-24 16:21:30,864 - DEBUG - ci_test]: (api_response) : Replica groups:
- Group: <group_id:432742929234133176>, <group_nodes:(9, 10)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_34_432744296733736961v0>, <shard_leader:10>, <shard_nodes:[9]>, Shard: <channel_name:by-dev-rootcoord-dml_35_432744296733736961v1>, <shard_leader:11>, <shard_nodes:[10]>]......  (api_request.py:27)
[2022-04-24 16:21:30,864 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-24 16:21:32,086 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-24 16:21:32,086 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 2, 20], kwargs: {} (api_request.py:55)
[2022-04-24 16:21:33,260 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-24 16:21:33,260 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', None, None, 20], kwargs: {} (api_request.py:55)
[2022-04-24 16:21:33,293 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 752, in query
    res = conn.query(self._name, expr, output_fields, partition_names, timeout, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 1002, in query
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=no shard leaders available for channel: by-dev-rootcoord-dml_35_432744296733736961v1, leaders: [11], err: fail to Query, QueryNode ID = 11, reason=collection hasn't been loaded or has been released, collection id = 432744296733736961)>
 (api_request.py:35)
[2022-04-24 16:21:33,293 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=no shard leaders available for channel: by-dev-rootcoord-dml_35_432744296733736961v1, leaders: [11], err: fail to Query, QueryNode ID = 11, reason=collection hasn't been loaded or has been released, collection id = 432744296733736961)> (api_request.py:36)
FAILED
```
**Server logs:**
[milvus_logs.tar.gz](https://github.com/milvus-io/milvus/files/8552750/milvus_logs.tar.gz)

### Expected Behavior

1. It should give an exception when loading with 2 replicas without release for the first time
2. After replica number update to 2, query successfully

**Comment by congqixia** [comment]
One occurrence was https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-ha-ci/detail/PR-16624/4/pipeline/, in which querynode panicked.
And proxy send search request to newly created querynode.

**Comment by ThreadDao** [comment]
image 'master-20220426-60f7fef3'
```
[2022-04-27 16:50:43,937 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 432812705375584257)  (api_request.py:27)
[2022-04-27 16:50:47,143 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:48,298 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-27 16:50:48,299 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', None, None, 20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:48,607 - DEBUG - ci_test]: (api_response) : [{'int64': 0}]  (api_request.py:27)
[2022-04-27 16:50:48,608 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 2, 20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:48,651 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-27 16:50:48,652 - DEBUG - ci_test]: (api_request)  : [Collection.get_replicas] args: [20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:48,687 - DEBUG - ci_test]: (api_response) : Replica groups:
- Group: <group_id:432812464710090829>, <group_nodes:(8, 11, 12, 9, 10)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_5_432812705047904257v1>, <shard_leader:8>, <shard_nodes:[8]>, Shard: <channel_name:by-dev-rootcoord-dml_4_432812705047904257v0>, <shard_leader:9>, <shard_nodes......  (api_request.py:27)
[2022-04-27 16:50:48,687 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-27 16:50:49,805 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-27 16:50:49,805 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 2, 20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:50,980 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-27 16:50:50,981 - DEBUG - ci_test]: (api_request)  : [Collection.get_replicas] args: [20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:51,018 - DEBUG - ci_test]: (api_response) : Replica groups:
- Group: <group_id:432812464710090846>, <group_nodes:(11, 9, 10)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_4_432812705047904257v0>, <shard_leader:9>, <shard_nodes:[9, 11]>, Shard: <channel_name:by-dev-rootcoord-dml_5_432812705047904257v1>, <shard_leader:10>, <shard_nodes:[......  (api_request.py:27)
[2022-04-27 16:50:51,018 - DEBUG - ci_test]: Replica groups:
- Group: <group_id:432812464710090846>, <group_nodes:(11, 9, 10)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_4_432812705047904257v0>, <shard_leader:9>, <shard_nodes:[9, 11]>, Shard: <channel_name:by-dev-rootcoord-dml_5_432812705047904257v1>, <shard_leader:10>, <shard_nodes:[10, 11]>]>
- Group: <group_id:432812464710090847>, <group_nodes:(8, 12)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_4_432812705047904257v0>, <shard_leader:12>, <shard_nodes:[12, 8]>, Shard: <channel_name:by-dev-rootcoord-dml_5_432812705047904257v1>, <shard_leader:8>, <shard_nodes:[8]>]> (test_collection.py:2256)
[2022-04-27 16:50:51,019 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', None, None, 20], kwargs: {} (api_request.py:55)
[2022-04-27 16:50:51,203 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 752, in query
    res = conn.query(self._name, expr, output_fields, partition_names, timeout, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 1003, in query
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Query, QueryNode ID = 8, reason=SharcCluster for by-dev-rootcoord-dml_5_432812705047904257v1 replicaID 432812464710090847 is no available)>
 (api_request.py:35)
[2022-04-27 16:50:51,205 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Query, QueryNode ID = 8, reason=SharcCluster for by-dev-rootcoord-dml_5_432812705047904257v1 replicaID 432812464710090847 is no available)> (api_request.py:36)
FAILED
```

**Comment by ThreadDao** [comment]
Looks fixed `master-20220428-68b1b82f`
